packages/protocol/test-sol/common/FeeCurrencyDirectory.t.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.7 <0.8.20;
+
+import "celo-foundry-8/Test.sol";
+import "../../contracts-0.8/common/FeeCurrencyDirectory.sol";
+
+contract MockOracle is IOracle {
+  uint256 numerator;
+  uint256 denominator;
+  uint256 lastUpdateTimestamp;
+
+  function setExchangeRate(uint256 _numerator, uint256 _denominator) public {
+    numerator = _numerator;
+    denominator = _denominator;
+    lastUpdateTimestamp = block.timestamp;
+  }
+
+  function getExchangeRateFor(address) external view returns (uint256, uint256, uint256) {
+    return (numerator, denominator, lastUpdateTimestamp);
+  }
+}
+
+contract FeeCurrencyDirectoryTestBase is Test {
```

Comment:
Would try to stick to the test contract naming convention adopted by [this PR](https://github.com/celo-org/celo-monorepo/pull/10870/files#diff-6a351948faad034f37124e48bc982850b280ea0269d4b9f76fcb57706658b313)